src/components/searchBar/SearchBar.jsx

```diff
@@ -0,0 +1,68 @@
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React, { useState } from "react";
+import "./SearchBar.css";
+
+const SearchBar = () => {
+  const [location, setLocation] = useState("");
+  const [checkIn, setCheckIn] = useState("");
+  const [checkOut, setCheckOut] = useState("");
+  const [guests, setGuests] = useState("");
+
+  const handleSearch = () => {
```

Comment:
Imagine we would want to reuse this component for different purposes. Let's say searching for home and the second use case would be searching for events. In such case we wouldn't want to implement the behavior what happens here in the component but in the place where we are calling this component. The good news is that we can pass this function via a prop. You can read about props in React here https://react.dev/learn/passing-props-to-a-component
Feel free to ask questions either here or in our slack channel.

Reply:
Thank you for the nice advise.
i updated the component to make it more reusable (maybe :) ). hope i got it and implemented well